controllers/algae_measurement.py

```diff
@@ -77,9 +78,25 @@ def measuring(self, initial_call) -> None:
 
             self.next_state("calculating")
 
-    @timed_state(duration=0.5, must_finish=True)
-    def recovering(self) -> None:
-        self.injector_component.desired_injector_voltage = -2.0
+    @state(must_finish=True)
+    def recovering(self, initial_call, state_tm) -> None:
+        if self.retract(initial_call, state_tm):
+            self.done()
+
+    def retract(self, initial_call, state_tm) -> bool:
+        self.injector_component.desired_injector_voltage = self.retraction_voltage
+        if initial_call:
+            self._tm = state_tm
```

Comment:
`state_tm` will be 0 on `initial_call`

Reply:
Not according to the docs?

`state_tm` - The number of seconds since this state has been active (note: it may not start at zero!)

https://robotpy.readthedocs.io/projects/utilities/en/latest/magicbot.html#module-magicbot.state_machine

Reply:
Okay, technically yes the first call will probably get a small non-zero `state_tm`